src/assets/assets.cpp

```diff
@@ -1209,6 +1207,41 @@ bool CAssetsCache::GetAssetIfExists(const std::string& name, CNewAsset& asset)
     return false;
 }
 
+bool GetAssetFromCoin(const Coin& coin, std::string& strName, CAmount& nAmount)
+{
+    if (!coin.IsAsset())
+        return false;
+
+    // Determine the type of asset that the scriptpubkey contains and return the name and amount
+    if (coin.out.scriptPubKey.IsNewAsset()) {
+        CNewAsset asset;
+        std::string address;
+        if (!AssetFromScript(coin.out.scriptPubKey, asset, address))
+            return false;
+        strName = asset.strName;
+        nAmount = asset.nAmount;
+        return true;
+    } else if (coin.out.scriptPubKey.IsTransferAsset()) {
+        CAssetTransfer asset;
+        std::string address;
+        if (!TransferAssetFromScript(coin.out.scriptPubKey, asset, address))
+            return false;
+        strName = asset.strName;
+        nAmount = asset.nAmount;
+        return true;
+    } else if (coin.out.scriptPubKey.IsOwnerAsset()) {
+        std::string name;
+        std::string address;
+        if (!OwnerAssetFromScript(coin.out.scriptPubKey, name, address))
+            return false;
+        strName = name;
+        nAmount = 1 * COIN;
```

Comment:
I notice that `1 * COIN` is used in a few places as a special value for owner assets.  Could be put in a constant somewhere as e.g. OWNER_ASSET_AMOUNT.